.github/scripts/add_release_label_to_pr_and_linked_issues.ts

```diff
@@ -0,0 +1,370 @@
+  }
+
+  return pullRequest;
+}
+
+// This function retrieves the timeline events for a pull request
+async function retrieveTimelineEvents(octokit: InstanceType<typeof GitHub>, repoOwner: string, repoName: string, prNumber: number): Promise<Array<Record<string, any>>> {
+
+  // We assume there won't be more than 100 timeline events
+  const retrieveTimelineEventsQuery = `
+    query($repoOwner: String!, $repoName: String!, $prNumber: Int!) {
+      repository(owner: $repoOwner, name: $repoName) {
+        pullRequest(number: $prNumber) {
+          timelineItems(itemTypes: [CONNECTED_EVENT, DISCONNECTED_EVENT], first: 100) {
+            nodes {
+              ... on ConnectedEvent {
+                __typename
+                createdAt
+                subject {
+                  ... on Issue {
+                    number
+                    id
+                    repository {
+                      name
+                      owner {
+                        login
+                      }
+                    }
+                  }
+                }
+              }
+              ... on DisconnectedEvent {
+                __typename
+                createdAt
+                subject {
+                  ... on Issue {
+                    number
+                    id
+                    repository {
+                      name
+                      owner {
+                        login
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  const retrieveTimelineEventsResult: {
+    repository: {
+      pullRequest: {
+        timelineItems: {
+          nodes: Array<{
+            __typename: 'ConnectedEvent' | 'DisconnectedEvent';
+            createdAt: string;
+            subject: {
+              number: number;
+              id: string;
+              repository: {
+                name: string;
+                owner: {
+                  login: string;
+                };
+              };
+            };
+          }>;
+        };
+      };
+    };
+  } = await octokit.graphql(retrieveTimelineEventsQuery, {
+    repoOwner,
+    repoName,
+    prNumber,
+  });
+
+  const timelineEvents = retrieveTimelineEventsResult?.repository?.pullRequest?.timelineItems?.nodes;
+
+  return timelineEvents;
+}
+
+// This function retrieves the list of linked issues for a pull request
+async function retrieveLinkedIssues(octokit: InstanceType<typeof GitHub>, repoOwner: string, repoName: string, prNumber: number): Promise<Labelable[]> {
+
+  // The list of linked issues can be deduced from timeline events
+  const timelineEvents = await retrieveTimelineEvents(octokit, repoOwner, repoName, prNumber);
+
+  const linkedIssuesMap: Record<string, Labelable> = {};
+
+  // This way to retrieve linked issues is not straightforward, but there's currently no easier way to obtain linked issues thanks to Github APIs
```

Comment:
Could we use `closingIssuesReferences` instead (see https://docs.github.com/en/graphql/reference/objects)? It seems to be the same thing as this

Reply:
This is now fixed with this [commit](https://github.com/MetaMask/metamask-extension/pull/19061/commits/cf2ed212bafb0f950e5e5b937136e39dcddcd9c4).